web-api/src/personalization/routes.rs

```diff
@@ -93,40 +95,28 @@ pub(crate) async fn update_interactions(
     user_id: &UserId,
     interactions: &[UserInteractionData],
 ) -> Result<(), Error> {
-    storage::Interaction::user_seen(storage, user_id).await?;
+    storage.user_seen(user_id).await?;
 
-    let ids = interactions
-        .iter()
-        .map(|document| &document.document_id)
-        .collect_vec();
-    let documents = storage::Document::get_by_ids(storage, &ids).await?;
-    let documents = documents
+    #[allow(clippy::zero_sized_map_values)]
+    let interaction_map = interactions
         .iter()
-        .map(|document| (&document.id, document))
+        .map(|interaction| (&interaction.document_id, interaction.interaction_type))
         .collect::<HashMap<_, _>>();
```

Comment:
Hm, maybe this needs to change.

I don't know for which reason we added support for passing multiple interactions but if it's batching we either should only take the last one (under the assumption that the user e.g. mistakenly clicked on like and then unliked again) or should handle all of them.

I don't think we should reject multiple interactions as it would make the API harder to use.